r/R/matrix.R

```diff
@@ -1109,15 +1109,15 @@ setMethod("rbind2", signature(x = "IterableMatrix", y = "IterableMatrix"), funct
   if (x@transpose != y@transpose) stop("Cannot merge matrices with different internal transpose states.\nPlease use transpose_storage_order().")
   if (matrix_type(x) != matrix_type(y)) {
     # upcast if mismatching types
-    type_order <- c("uint32_t", "float", "double")
-    new_type <- type_order[max(match(matrix_type(x), type_order), match(matrix_type(y), type_order))]
     rlang::warn(
-      sprintf("First matrix type: %s, second matrix type: %s.\nUpcasting to %s to match matrix types", 
-        matrix_type(x), matrix_type(y), new_type
+      sprintf(
+        paste0("Mismatching matrix types when calling %sbind().\n",
+        "First matrix type: %s, second matrix type: %s.\nUpcasting to double to match matrix types"), 
```

Comment:
I'm realizing this warning might get printed multiple times if someone is calling `rbind` or `cbind` to merge many matrices at once. Maybe we can shorten it to one line: `"%sbind2(): Mismatching matrix types (%s vs. %s). Upcasting to double`

As a side note, `rlang::warn()` supports character vectors for its first arg in which case it makes nicely bulleted output (better than manual newlines). But in this case it's probably best to just squeeze on one line.